directory_sso_api_client/user.py

```diff
@@ -61,22 +71,31 @@ def get_last_login(self, start=None, end=None):
 
         return self.get(
             url=self.endpoints['last_login'],
-            params=params
+            params=params,
+            authenticator=authenticator,
         )
 
-    def create_user(self, email, password):
+    def create_user(self, email, password, authenticator=None):
         url = self.endpoints['user_create']
         data = OrderedDict(
             [('email', email), ('password', password)]
         )
-        return self.post(url, data)
+        return self.post(
+            url,
+            data,
+            authenticator=authenticator,
+        )
 
-    def create_user_profile(self, sso_session_id, data):
-        authenticator = AuthenticatorNegotiator(sso_session_id=sso_session_id)
+    def create_user_profile(self, sso_session_id, data, authenticator=None):
+        authenticator = authenticator or AuthenticatorNegotiator(
+            sso_session_id=sso_session_id
+        )
         url = self.endpoints['user_create_profile']
         return self.post(url, data, authenticator=authenticator)
 
-    def update_user_profile(self, sso_session_id, data):
-        authenticator = AuthenticatorNegotiator(sso_session_id=sso_session_id)
+    def update_user_profile(self, sso_session_id, data, authenticator=None):
```

Comment:
`def update_user_profile(self, sso_session_id, data, authenticator=AuthenticatorNegotiator)`:

Reply:
After thinking about it for a while I realised that neither solution is going to work properly. 
It's because these 3 methods: `get_oauth2_user_profile()` `create_user_profile()`, `update_user_profile()` already use custom authenticators which are required to access underlying API endpoints. 
By replacing them with another custom authenticator e.g. Basic Auth one, we won't be able to properly access such endpoints.

Reply:
I'm going to restore original logic to those 3 methods.